Transient dev mode error on Windows

We often have this error on Windows:
```
[ERROR] Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 12.488 s <<< FAILURE! - in io.quarkus.resteasy.test.NotFoundExceptionMapperTestCase
[ERROR] shouldNotDisplayDeletedFileIn404ErrorPage  Time elapsed: 2.532 s  <<< ERROR!
java.io.UncheckedIOException: 
java.nio.file.FileSystemException: D:\a\quarkus\quarkus\extensions\resteasy\deployment\target\quarkus-dev-mode-test7761304085196119552\src\main\resources\META-INF\resources\test.html: The process cannot access the file because it is being used by another process.

	at io.quarkus.test.QuarkusDevModeTest.deleteResourceFile(QuarkusDevModeTest.java:409)
	at io.quarkus.resteasy.test.NotFoundExceptionMapperTestCase.shouldNotDisplayDeletedFileIn404ErrorPage(NotFoundExceptionMapperTestCase.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:686)
```

--- a/test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java
+++ b/test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java
@@ -404,10 +404,27 @@ public class QuarkusDevModeTest
      */
     public void deleteResourceFile(String path) {
         final Path resourceFilePath = deploymentResourcePath.resolve(path);
-        try {
-            Files.delete(resourceFilePath);
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
+        long timeout = System.currentTimeMillis() + 5000;
+        //in general there is a potential race here
+        //if you serve a file you will send the data to the client, then close the resource
+        //this means that by the time the client request is run the file may not 
+        //have been closed yet, as the test sees the response as being complete after the last data is send
+        //we wait up to 5s for this condition to be resolved
+        for (;;) {
+            try {
+                Files.delete(resourceFilePath);
+                break;
+            } catch (IOException e) {
+                try {
+                    Thread.sleep(50);
+                } catch (InterruptedException ex) {
+                    //ignore
+                }
+                if (System.currentTimeMillis() < timeout) {
+                    continue;
+                }
+                throw new UncheckedIOException(e);
+            }
         }
         // wait for last modified time of the parent to get updated
         sleepForFileChanges(resourceFilePath.getParent());

